juju/juju
# [JUJU-189] Raft lease extension timeout

The following changes allow extensions on leases to have a larger
timeout when attempting to enqueue on to the raft lease queue. The aim
behind this change is to allow less churn for a given application lease.

During load testing, it is seen that when claims block the queue all
subsequent extensions are also thrown away. This is highly problematic
as it causes application leader changes, which also cause more churn in
the database. The change here is to be more forgiving for extensions and
hope after a lost claim, we get to push some extensions through so we
can handle more throughput.


## QA steps

See for Q&A steps for a larger in-depth test https://github.com/juju/juju/pull/13453

```
$ juju bootstrap lxd test1 --build-agent --config features="[raft-api-leases,raft-batch-fsm]"
$ juju model-config -m controller logging-config="<root>=INFO;juju.worker.lease.raft=TRACE;juju.core.raftlease=TRACE;juju.worker.globalclockupdater.raft=TRACE;juju.worker.raft=TRACE"
$ juju debug-log -m controller
```

Ensure there are no errors.
